vim_menutrans_helper.py

```diff
@@ -0,0 +1,238 @@
+#!/user/bin/env python3
+
+import re
+import glob
+import os
+import sys
+
+menu_commands = {"me", "menu", "noreme", "noremenu", "am",
+             "amenu", "an", "anoremenu", "nme", "nmenu",
+             "nnoreme", "nnoremenu", "ome", "omenu", "onoreme",
+             "onoremenu", "vme", "vmenu", "vnoreme", "vnoremenu",
+             "xme", "xmenu", "xnoreme", "xnoremenu", "sme", "smenu",
+             "snoreme", "snoremenu", "ime", "imenu", "inoreme", "inoremenu",
+             "cme", "cmenu", "cnoreme", "cnoremenu", "tlm", "tlmenu", "tln",
+             "tlnoremenu", "tunmenu", "tu", "popup", "popu"}
+toolbar_commands = {"tmenu", "tm"}
+exe_commands = {"exe", "exec", "execute"}
+sil_commands = {"sil", "silent", "sil!", "silent!"}
+translated_menu = {"tmenu", "menut", "menutrans", "menutranslate"}
+
+def unescape_double_quotes(origin):
+    """Strip the quotes and unescape a string inside double quotes."""
+
+    return (origin[1:-1].replace('\\"', '"')
+                        .replace("\\'", "'")
+                        .replace('\\t', '\t')
+                        .replace('\\\\', '\\'))
+
+def unescape_single_quotes(origin):
+    """Strip the quotes and unescape a string inside single quotes."""
+
+    return origin[1:-1].replace("''", "'")
+
+def make_untranslated_dict(untranslation_file, untranslated_dict):
+    """Get the diffence between untranslated file and translated file."""
+    
+    # Get the untranslated files' content
+    # Encode with "latin1", because we don't care about the correctness of 
+    # non-english character
+    with open(untranslation_file, encoding="latin1") as f1:
+        for line_number, line in enumerate(f1):
+            to_be_translated_words = []
+
+            line = line.strip()
+
+            # Skip the Comments
+            if line.startswith('"'):
+                continue
+            quoted_token_list = []
+            pre_end = -1
+            top_string = line.split()
+            if len(top_string) > 0 and top_string[0] in exe_commands:
+                # Extract content inside quotes from line
+                for match in re.finditer(
+                        r"'(?:''|[^'])*'|(?:\"(?:(?:\\.)|[^\"])*\")", line):
+                    mid_part = line[pre_end + 1: match.start()].strip()
+
+                    # Add a digit, namely "5", if there are variables inside 
+                    # mid_part
+                    if not ((len(mid_part) == 1 and mid_part[0] == ".") or 
```

Comment:
`(len(mid_part) == 1 and mid_part[0] == ".")` is just equivalent to `mid_part == "."`. You don't need to first check the string length then verify the only character.

However, as single-dot string concatenation has been deprecated by Vim in favor of double dots: `..`, you'll have to support both:

```python
if not (mid_part == '.' or mid_part == '..' or
```